packages/python-packages/apiview-copilot/evals/_custom.py

```diff
@@ -201,3 +328,547 @@ def __call__(self, *, response: str, query: str, language: str, actual: str, tes
+                
+                run_summary["accuracy"] = (run_summary["correct_actions"] / run_summary["total_tests"]) * 100 if run_summary["total_tests"] > 0 else 0
+                all_results[file_name].append(run_summary)
+        
+        # For multiple runs: take the median accuracy run
+        final_results = {}
+        for file_name, runs in all_results.items():
+            if len(runs) == 1:
+                final_results[file_name] = runs[0]
+            else:
+                # Find median by accuracy
+                sorted_runs = sorted(runs, key=lambda x: x["accuracy"])
+                median_idx = len(sorted_runs) // 2
+                final_results[file_name] = sorted_runs[median_idx]
+        
+        return final_results
+    
+    def show_results(self, processed_results: dict) -> None:
+        """Display thread-resolution workflow results with action breakdown."""
+        from tabulate import tabulate
+        
+        for file_name, results in processed_results.items():
+            accuracy = results["accuracy"]
+            correct = results["correct_actions"]
+            total = results["total_tests"]
+            
+            print("====================================================")
+            print(f"\n\n✨ {file_name} results:\n")
+            print(f"Overall Accuracy: {correct}/{total} ({accuracy:.1f}%)\n")
+            
+            # Show breakdown by action type
+            breakdown = results["action_breakdown"]
+            breakdown_rows = []
+            for action_type, stats in breakdown.items():
+                if stats["total"] > 0:
+                    type_accuracy = (stats["correct"] / stats["total"]) * 100
+                    breakdown_rows.append([
+                        action_type,
+                        f"{stats['correct']}/{stats['total']}",
+                        f"{type_accuracy:.1f}%"
+                    ])
+            
+            if breakdown_rows:
+                print("Action Type Breakdown:")
+                print(tabulate(breakdown_rows, 
+                             headers=["Action Type", "Correct", "Accuracy"], 
+                             tablefmt="simple"))
+                print()
+            
+            # Show failed cases with details
+            failed_cases = [test for test in results["test_results"] if not test["correct"]]
+            if failed_cases:
+                print("Failed Cases:")
+                for test in failed_cases:
+                    print(f"  ❌ {test['testcase']}")
+                    print(f"     Expected: {test['expected_action']}")
+                    print(f"     Actual:   {test['actual_action']}")
+                    print()
+
+
+# Registry of available evaluators
+EVALUATORS: Dict[str, BaseEvaluator] = {
+    "apiview": CustomAPIViewEvaluator,
+    "mention-action": MentionActionEvaluator,
+    "thread-resolution-action": ThreadResolutionActionEvaluator
```

Comment:
What are the actual differences between these two evaluators? It seems like they would be effectively or entirely the same. In this instance the goal would be to reuse the same class (`PromptyEvaluator` for example) for both of these, with any differences expressed via the config.